### PR TITLE
Fix owners of AsFunction applied expressions

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/core/quoted/PickledQuotes.scala [hunk 1]
```diff
@@ -143,7 +143,7 @@ object PickledQuotes {
             case tree: Ident if tree.symbol == paramSym => x1Ref().withPos(tree.pos)
             case _ => super.transform(tree)
           }
-        }.transform(ddef.rhs)
+        }.transform(ddef.rhs).changeOwner(ddef.symbol, ctx.owner)
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/core/quoted/PickledQuotes.scala]: To be more efficient, you can replace your TreeMap by a TreeTypeMap(oldOwners = ..., newOwners = ...), and avoid the need for changeOwner (which would create its own TreeTypeMap)

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/core/quoted/PickledQuotes.scala]: Will change it

**Comment by nicolasstucki** [comment]
Tests are failing on maser :( 

@odersky told me he already has a fix for it.

**Comment by Blaisorblade** [comment]
Cherry-picked likely fix in #4141.

**Comment by Blaisorblade** [comment]
And new build at http://dotty-ci.epfl.ch/lampepfl/dotty/3824

**Pull request merged**